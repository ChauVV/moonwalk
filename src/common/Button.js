import React from 'react';
import styled from 'styled-components';
import LinearGradient from 'react-native-linear-gradient';
import { TouchableOpacity } from 'react-native';

const TouchableWrapper = styled.TouchableOpacity`
    shadow-opacity: 0.75;
    shadow-radius: 5px;
    shadow-color: ${({shadow}) => shadow};
    shadow-offset: 0px 0px;
`;

const Wrapper = styled(LinearGradient)`
    border-radius: 10px;
    padding: 15px 30px;
    margin: 0 5px;
`;

const Title = styled.Text`
    color: #efefef;
    font-size: 20px;
    font-weight: bold;
`;

const TYPES = {
    primary: {
        gradient: ['#6b2afd', '#5e15ea'],
        shadow: '#5e15ea',
    },
    secondary: {
        gradient: ['#3a3a63', '#2c2f53'],
        shadow: '#2c2f53',
    }
}

export default ({title, onPress, type = 'primary'}) => (
    <TouchableWrapper onPress={onPress} shadow={TYPES[type].shadow}>
        <Wrapper
            colors={TYPES[type].gradient}
            start={{x: 0.0, y: 0.0}} 
            end={{x: 1.0, y: 0.0}}
        >
            <Title>{title}</Title>
        </Wrapper>
    </TouchableWrapper>
)