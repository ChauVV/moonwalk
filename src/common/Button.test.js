import React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with title correctly", () => {
  const tree = renderer.create(<Button title="Press me" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with a different type style", () => {
  const tree = renderer
    .create(<Button type="secondary" title="Secondary" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
