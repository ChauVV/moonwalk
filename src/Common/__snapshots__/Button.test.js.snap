// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "shadowColor": "#5e15ea",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.75,
      "shadowRadius": 5,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4285213437,
        4284356074,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 10,
          "marginBottom": 0,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 0,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#efefef",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    />
  </BVLinearGradient>
</View>
`;

exports[`renders with a different type style 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "shadowColor": "#2c2f53",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.75,
      "shadowRadius": 5,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4282006115,
        4281085779,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 10,
          "marginBottom": 0,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 0,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#efefef",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Secondary
    </Text>
  </BVLinearGradient>
</View>
`;

exports[`renders with title correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "shadowColor": "#5e15ea",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.75,
      "shadowRadius": 5,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4285213437,
        4284356074,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 10,
          "marginBottom": 0,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 0,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#efefef",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Press me
    </Text>
  </BVLinearGradient>
</View>
`;
